firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java

```diff
@@ -404,26 +404,31 @@ private void logAppStartTrace() {
             .setDurationUs(getClassLoadTimeCompat().getDurationMicros(onResumeTime));
     List<TraceMetric> subtraces = new ArrayList<>(/* initialCapacity= */ 3);
 
-    TraceMetric.Builder traceMetricBuilder =
-        TraceMetric.newBuilder()
-            .setName(Constants.TraceNames.ON_CREATE_TRACE_NAME.toString())
-            .setClientStartTimeUs(getClassLoadTimeCompat().getMicros())
-            .setDurationUs(getClassLoadTimeCompat().getDurationMicros(onCreateTime));
-    subtraces.add(traceMetricBuilder.build());
-
-    traceMetricBuilder = TraceMetric.newBuilder();
-    traceMetricBuilder
-        .setName(Constants.TraceNames.ON_START_TRACE_NAME.toString())
-        .setClientStartTimeUs(onCreateTime.getMicros())
-        .setDurationUs(onCreateTime.getDurationMicros(onStartTime));
-    subtraces.add(traceMetricBuilder.build());
-
-    traceMetricBuilder = TraceMetric.newBuilder();
-    traceMetricBuilder
-        .setName(Constants.TraceNames.ON_RESUME_TRACE_NAME.toString())
-        .setClientStartTimeUs(onStartTime.getMicros())
-        .setDurationUs(onStartTime.getDurationMicros(onResumeTime));
-    subtraces.add(traceMetricBuilder.build());
+    TraceMetric.Builder traceMetricBuilder = TraceMetric.newBuilder()
+              .setName(Constants.TraceNames.ON_CREATE_TRACE_NAME.toString())
+              .setClientStartTimeUs(getClassLoadTimeCompat().getMicros())
+              .setDurationUs(getClassLoadTimeCompat().getDurationMicros(onCreateTime));
+      subtraces.add(traceMetricBuilder.build());
+
+    // OnStartTime is not captured in all situations, so checking for valid value before using it.
+    if (onStartTime != null) {
+      traceMetricBuilder = TraceMetric.newBuilder();
+      traceMetricBuilder
+          .setName(Constants.TraceNames.ON_START_TRACE_NAME.toString())
+          .setClientStartTimeUs(onCreateTime.getMicros())
+          .setDurationUs(onCreateTime.getDurationMicros(onStartTime));
+      subtraces.add(traceMetricBuilder.build());
+    }
+
+    // onResumeTime is not captured in all situations, so checking for valid value before using it.
+    if (onResumeTime != null) {
+      traceMetricBuilder = TraceMetric.newBuilder();
+      traceMetricBuilder
+          .setName(Constants.TraceNames.ON_RESUME_TRACE_NAME.toString())
+          .setClientStartTimeUs(onStartTime.getMicros())
+          .setDurationUs(onStartTime.getDurationMicros(onResumeTime));
+      subtraces.add(traceMetricBuilder.build());
+    }
```

Comment:
Move this under the `onStartTime != null` if statement since this depends on `onStartTime` not being null.

Reply:
Done.